docs/plugins/plugin-writer/first-plugin.rst

```diff
@@ -1,2 +1,106 @@
 Writing Your First Plugin
 =========================
+
+Understanding models
+--------------------
+There are models which are expected to be used in plugin implementation, so understanding what they are designed for is useful for a plugin writer.
+Each model below have a link to its documentation where its purpose, all attributes and relations are listed.
+
+Here is a gist of how models are related to each other and what each model is responsible for.
+
+* :class:`~pulpcore.app.models.Repository` contains :class:`~pulpcore.plugin.models.Content`. :class:`~pulpcore.plugin.models.RepositoryContent` is used to represent this relation.
+* :class:`~pulpcore.plugin.models.Content` can have :class:`~pulpcore.plugin.models.Artifact` associated with it. :class:`~pulpcore.plugin.models.ContentArtifact` is used to represent this relation.
+* :class:`~pulpcore.plugin.models.ContentArtifact` can have :class:`~pulpcore.plugin.models.RemoteArtifact` associated with it.
+* :class:`~pulpcore.plugin.models.Artifact` is a file.
+* :class:`~pulpcore.plugin.models.RemoteArtifact` contains information about :class:`~pulpcore.plugin.models.Artifact` from a remote source, including URL to perform download later at any point.
+* :class:`~pulpcore.plugin.models.Importer` knows specifics of the plugin :class:`~pulpcore.plugin.models.Content` to put it into Pulp. :class:`~pulpcore.plugin.models.Importer` defines how to synchronize remote content. Pulp Platform provides two implementations for concurrent downloads of remote content: :ref:`concurrent.futures <futures-docs>` and :ref:`asyncio <asyncio-docs>` approaches. Plugin writer is encouraged to use one of them but is not required to.
+* :class:`~pulpcore.plugin.models.PublishedArtifact` refers to :class:`~pulpcore.plugin.models.ContentArtifact` which is published and belongs to a certain :class:`~pulpcore.app.models.Publication`.
+* :class:`~pulpcore.plugin.models.PublishedMetadata` is a repository metadata which is published, located in ``/var/lib/pulp/published`` and belongs to a certain :class:`~pulpcore.app.models.Publication`.
+* :class:`~pulpcore.plugin.models.Publisher` knows specifics of the plugin :class:`~pulpcore.plugin.models.Content` to make it available outside of Pulp. :class:`~pulpcore.plugin.models.Publisher` defines how to publish content available in Pulp.
+* :class:`~pulpcore.app.models.Publication` is a result of publish operation of a specific :class:`~pulpcore.plugin.models.Publisher`.
```

Comment:
Need to document :class:`~pulpcore.app.models.Distribution`

Each publisher has associated distributions which define how a publication is distributed.

Reply:
The initial idea was to cover only models which plugin writers may use at some point.
But I already mentioned `Publication` so I guess it makes sense to add `Distribution` as well. Will do.